cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/WorkflowHttpHandler.java

```diff
@@ -100,8 +100,11 @@ public void suspendWorkflowRun(HttpRequest request, final HttpResponder responde
       }
       controller.suspend().get();
       responder.sendString(HttpResponseStatus.OK, "Program run suspended.");
-    }  catch (SecurityException e) {
+    } catch (SecurityException e) {
       responder.sendStatus(HttpResponseStatus.UNAUTHORIZED);
+    } catch (NotFoundException e) {
+      LOG.warn("NotFoundException while runtime information for run id is not found.", e);
+      responder.sendString(HttpResponseStatus.NOT_FOUND, e.getMessage());
     } catch (Throwable e) {
```

Comment:
You can remove these catch-blocks (SecurityException, NotFoundException, Throwable) since HttpExceptionHandler is handling them already.